### ISSUE TITLE
[Bug] Bookmarks and search results fail to open in new tabs when current tab is in container.

### ISSUE THREAD

**Issue by mike-tn** [piroor/treestyletab#2360]
## Short description
Bookmarks ans search results fail to open in a new tab when current tab has some site loaded (is non-new/blank) in any container, and some TST and Firefox preferences are set.

## Steps to reproduce

 1. Prepare Firefox version 68.0.1 64bit. (In fact I have FirefoxPortable but VERY UNLIKELY it matters here)
 2. Start Firefox with clean profile.
 3. In Firefox options set Home/New tabs = Blank Page
 4. in about:config set browser.tabs.loadBookmarksInTabs=true (without this, bookmarks bug doesn't happen but search results are still buggy)
 5. Install TST 3.1.7.
 6. In TST options set 
       Open new blank tab as  = a child of a current tab
       Guess a new opened tab ..... = about:blank
       Check  Inherit the container of the parent tab to a new blank tab.
 7. Add a new tab in Personal container via plus button on the Tabs bar on top of Firefox.
 8. Load github.com (on any other site)
 9 . On a firefox toolbar click  Library button, choose  Bookmarks and click any preset bookmark (like Mozilla's 'About Us' or 'PortableApps.com').

## Expected result
Selected bookmark opened in a new tab as a child of the current tab in the Personal container.

## Actual result
A blank active tab is opened as a child of the current tab in the Personal container. And no Active/selected tab in the TST sidebar!

10. In the tab with site opened in some container type "test" in location bar and press ALT-Enter. (not needed to set browser.tabs.loadBookmarksInTabs=true , maybe set to default value)

## Expected result
Google search results opened in a new tab as a child of the current tab in the same container.

## Actual result
A blank active tab is opened as a child of the current tab in the same container. And no Active/selected tab in the TST sidebar.

Sometimes the bug doesn't happen (bookmark opens) but not often. I have the bug always reproduced in this situation: open blank tab in a container, open bookmark on it, and then try to open the same bookmark again.

## Environment

 * Platform (OS): Windows 7 Pro 64bit (8gb)
 * Version of Firefox: 68.0.1 Firefox Portable 64bit
 * Version (or revision) of Tree Style Tab: 3.1.7

**Comment by mike-tn** [comment]
Just checked all this on fresh installation of Firefox 68.0.2 64bit (NOT portable) - the same bug!!!

**Comment by piroor** [comment]
Hmm... it looks quite edge case problem.

Basically TST is not designed to apply the container of the active tab for any new tab, instead applies only for new tabs opened with the "+" button or Ctrl-T. "Guess a new opened tab..." option is provided for cases when you use custom new tab page changed by other addons. The URI "about:blank" is quite special and used on various cases Firefox tries to open tabs, so applying TST's special behavior for "about:blank" may produce odd behaviors like you saw.

On these case you wrote, I think that TST should not apply the container of the active tab for tabs opened from bookmarks or the location bar, because they are not new tabs opened by Ctrl-T or the +" button. On these cases the "Personal" container is applied just accidentally and loading of new tabs with URLs are accidentally canceled.

If you with to apply the container of the current tab for any tab opened from bookmarks or the location bar, I think you should use other addons which is designed to do that. (But sadly I don't know which addon has the feature, so I cannot list them here, sorry.)

**Comment by mike-tn** [comment]
Many  people use about:blank as a new tab.
Also many have settings in about:config for opening bookmarks and search results in new tabs.
For them options  "Open new blank tab as = a child of a current tab" and "Inherit the container of the parent tab to a new blank tab" would be incompatible.

I think it would be precious to many to be able  to open all tabs in a window or in a tree in one container, including bookmarks and search results.  It could be a big plus for the TST to bring such ability.
Using another add-on for that is very inconvenient, and reduces value of TST.

**Comment by mike-tn** [comment]
Is there any ways to open a new tab in the same container as in a current tab (without directly selecting container) using mouse with unchecked "Inherit the container of the parent tab to a new blank tab"?

**Comment by piroor** [comment]
Hmm, since it was implemented on Firefox I've been used `about:newtab`, so I didn't realized that `about:blank` new tab produces such a problem. I think I need to do something on this...

**Comment by mike-tn** [comment]
Thank you for spending your time on this!!!

**Comment by piroor** [comment]
With commits 6097fa0 and 75b81e4 now TST can detect the difference between a new blank tab and a placeholder blank tab opened for "Open in New Tab" command. Now TST should apply "Open new blank tab as = a child of a current tab" behavior only for new blank tabs and tabs opened for bookmarks won't be affected. Thus,

* At the step 9, the selected bookmark will be opened in a new tab as an independent tab in the default container. (It won't become a child of the current tab, and it won't inherit the container of the current tab.)
* At the step 10, the Alt-Enter-ed URL will be opened in a new tab as an independent tab in the default container. (It won't become a child of the current tab, and it won't inherit the container of the current tab.)
  * However, the Alt-Enter-ed URL will be opened in a new child tab with the inherited container "Personal" if the URL has the same origin as the current tab. This is the result of the feature "open new tab as a child of the current tab, when it is opened from the location bar or from outside Firefox with an URL having the same origin as the current tab".

In other words, TST still doesn't provide ability to open a new tab with the container same to the current tab, from bookmarks or from the location bar. You may need to do:

1. Open a new blank tab with non-default container.
2. Try to open a tab from the location bar or from bookmarks. Then Firefox automatically redirects the request to reuse the current tab regardless it is non-default container, if it is blank.

**Comment by piroor** [comment]
Open bookmark in Container Tab
https://addons.mozilla.org/firefox/addon/open-bookmark-in-container-tab/
I think this will help you to open a bookmark with non-default container.

**Comment by piroor** [comment]
I marked this issue as "wontfix" again because I still have no plan to introduce ability to apply the container of the current tab to new tabs opened with URL with different origin as the current tab.

**Comment by irvinm** [comment]
@mike-tn can you review and comment or close this item based on the information?  Thanks.

**Comment by mike-tn** [comment]
OK. I think we can close it for now.
